tests/testsuite/warn_on_failure.rs

```diff
@@ -86,31 +82,55 @@ fn no_warning_on_bin_failure() {
     upstream
         .cargo("build")
         .with_status(101)
-        .with_stdout_does_not_contain("hidden stdout")
-        .with_stderr_does_not_contain("hidden stderr")
-        .with_stderr_does_not_contain(&format!("[WARNING] {}", WARNING1))
-        .with_stderr_does_not_contain(&format!("[WARNING] {}", WARNING2))
-        .with_stderr_contains("[UPDATING] `[..]` index")
-        .with_stderr_contains("[DOWNLOADED] bar v0.0.1 ([..])")
-        .with_stderr_contains("[COMPILING] bar v0.0.1")
-        .with_stderr_contains("[COMPILING] foo v0.0.1 ([..])")
+        .with_stdout_data("")
+        .with_stderr_data(str![[r#"
+[UPDATING] `dummy-registry` index
+[LOCKING] 2 packages to latest compatible versions
+[DOWNLOADING] crates ...
+[DOWNLOADED] bar v0.0.1 (registry `dummy-registry`)
+[COMPILING] bar v0.0.1
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
+error[E0425]: cannot find function `hi` in this scope
+ --> src/main.rs:1:13
+  |
+1 | fn main() { hi() }
+  |             ^^ not found in this scope
+
+For more information about this error, try `rustc --explain E0425`.
+[ERROR] could not compile `foo` (bin "foo") due to 1 previous error
+
+"#]])
```

Comment:
I think in this case it works because if something changes and, the diff for the snapshot update will be suspicious, having "hidden" stuff showing up.